rust/src/applayertemplate/template.rs

```diff
@@ -342,10 +342,10 @@ pub extern "C" fn rs_template_parse_response(
     };
     let state = cast_pointer!(state, TemplateState);
     let buf = build_slice!(input, input_len as usize);
-    if state.parse_response(buf) {
-        return 1;
+    if !state.parse_response(buf) {
```

Comment:
Seems like it might be useful to do 

```
impl From<i32> for AppLayerReturn {
    fn from(v: i32) -> Self {
        if v < 0 {
            Self::err()
        } else {
            Self::ok()
        }
    }
}```

so that most of these just become `state.parse_response(buf).into()`

Reply:
Can you check my attempt here https://github.com/OISF/suricata/pull/4677?